File: librarymanager/librarymanager-be/src/main/java/ru/elenacherev/librarymanager/api/ReaderApi.java
package ru.elenacherev.librarymanager.api;

import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ru.elenacherev.librarymanager.api.dto.BookUsing;
import ru.elenacherev.librarymanager.api.dto.Order;
import ru.elenacherev.librarymanager.api.dto.Reader;

public interface ReaderApi {

	Reader getReader(
			@NotNull Long readerId
	);

	Page<Reader> searchReaders(
			Pageable pageable
	);

	Reader createReader(
			@Valid Reader reader
	);

	Reader saveReader(
			@NotNull Long readerId, 
			@Valid Reader reader
	);

	Page<Order> searchOrders(
			Long readerId, 
			Pageable pageable
	);

	List<Order> updateOrders(
			@NotNull Long readerId, 
			@NotNull List<Long> orderIds
	);

	Page<BookUsing> searchBookUsings(
			@NotNull Long readerId, 
			Pageable pageable
	);

	List<BookUsing> updateBookUsings(
			@NotNull Long readerId, 
			@NotNull List<Long> bookUsingIds
	);

}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/ReaderApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.BookUsing
import ru.elenacherev.librarymanager.api.dto.Order
import ru.elenacherev.librarymanager.api.dto.Reader
import java.util.*
import javax.validation.Valid

interface ReaderApi {

    fun getReader(
        readerId: UUID
    ): Reader?

    fun searchReaders(
        pageable: Pageable
    ): Page<Reader>

    fun createReader(
        reader: @Valid Reader
    ): Reader

    fun saveReader(
        readerId: UUID,
        reader: @Valid Reader
    ): Reader

    fun searchOrders(
        readerId: UUID,
        pageable: Pageable
    ): Page<Order>

    fun updateOrders(
        readerId: UUID,
        orderIds: List<UUID>
    ): List<Order>

    fun searchBookUsings(
        readerId: UUID,
        pageable: Pageable
    ): Page<BookUsing>

    fun updateBookUsings(
        readerId: UUID,
        bookUsingIds: List<UUID>
    ): List<BookUsing>
}